test/typescript/cross-stack-references/test.ts

```diff
@@ -317,6 +330,7 @@ describe("cross stack references", () => {
               "constructPath": "functionOutput",
               "workingDirectory": "stacks/functionOutput",
               "synthesizedStackPath": "stacks/functionOutput/cdk.tf.json",
+              "stackMetadataPath": "stacks/functionOutput/metadata.json",
```

Comment:
Is the metadata extracted into a new file in general now or only for HCL? I think in general wold make more sense, but we need to make sure telemetry knows about it (I think this is read somewhere 🤔 )

Reply:
So, this was indeed a problem with HCL generation. Because we read the entire stack back at times primarily to read metadata, I had to copy that metadata (and the full terraform block) to ensure that functionality doesn't break. I've made the part of the code load the stack content aware of the separate metadata file, so it should work as expected (from my understanding at least). And yes, this only needed for HCL.